### DISCUSSION TITLE
Why `kalosm_language::chat::Chat` consumes LLM?

### DISCUSSION THREAD

**Discussion by InAnYan** [floneum/floneum#295] [Q&A] [answered]
What if I want to have 2 chats? I have to load 2 LLMs?

In other LLM crates/(libraries from other programming languages), typically `Chat` is just a structure that holds messages and resembles and **API request**. It exists separately from LLM. And then LLM will consume that request and do what it needs to do. IMHO this is a better approach, as `Chat` doesn't depend on LLM and it's easy to modularize libraries and test them.

**Comment by ealmloff** [comment, marked as answer]
The `Llama` struct is a reference counted pointer to the language model. It is cheaply clonable so you can use it in many different places

**Reply by InAnYan** [reply]
Thanks for the clarification
